index.js

```diff
@@ -3,51 +3,61 @@ const inputTitle = document.querySelector('#title');
 const inputAuthor = document.querySelector('#author');
 const addBtn = document.querySelector('#add');
 
-const savedData = localStorage.getItem('savedInput');
+class Book {
+  constructor(savedData = []) {
+    this.arr = savedData;
+  }
 
-let collection = [];
+  saveData(data) {
+    let existing = JSON.parse(localStorage.getItem('book'));
+    existing = existing || [];
+    this.arr = existing;
+    this.arr.push(data);
+    localStorage.setItem('book', JSON.stringify(this.arr));
+  }
 
-if (savedData && savedData !== null) {
-  collection = JSON.parse(savedData);
-}
+  removeBook = (index) => {
+    if (index !== null && index !== undefined) {
+      this.arr.splice(index, 1);
+      localStorage.setItem('book', JSON.stringify(this.arr));
+      this.getData();
+    }
+  };
```

Comment:
- [ ] _[OPTIONAL]_ I would suggest using `arr.filter` or `arr.slice` instead of `arr.splice` since it mutates the array directly.

Reply:
> * [ ]  _[OPTIONAL]_ I would suggest using `arr.filter` or `arr.slice` instead of `arr.splice` since it mutates the array directly.

Okay @omar-labana  We'll have that sorted. Thank you very much